src/main/scala/scalaz/reactive/Future.scala

```diff
@@ -2,33 +2,44 @@ package scalaz.reactive
 
 import scalaz.Scalaz._
 import scalaz.reactive.Time.{ NegInf, PosInf }
+import scalaz.zio.IO
 import scalaz.{ Applicative, Functor, Monad, Monoid }
 
-case class Future[+A](time: () => Time, force: () => A) {
+object Types {
+  type Infallible[A] = IO[Nothing, A]
+}
+
+case class Future[+A](force: IO[Nothing, (Time, A)]) {
 
   def + [AA >: A](other: Future[AA]): Future[AA] =
-    Future(
-      () => time().min(other.time()),
-      () => if (time() <= other.time()) force() else other.force()
-    )
+    Future(force.flatMap {
+      case (t1, a1) =>
+        other.force.map {
+          case (t2, a2) => if (t1 <= t2) (t1, a1) else (t2, a2)
+        } //There should be IO.ap, is there?
```

Comment:
I think this has the same dilemma as the events... if we make Future[A] = IO[Void, (Time, A)] then I would say we shouldn't implement this monoid for now